### ISSUE TITLE
Prefabs and entities can be reparented to other prefabs using Transform component

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#15260]
**Describe the bug**
Currently reparenting entities or prefabs into another prefab should not be possible due to https://github.com/o3de/o3de/pull/14411. While doing so via drag and dropping is blocked, it is still possible to do this via *Transform* > *Parent entity* and manually picking the parent. Parenting entities/prefabs that way produces multiple warning and errors (please refer to the Editor.log in the **Video and Editor.log** section for detailed information).

It should be noted that parenting prefab to itself this way produces an assert (but is still blocked with the cyclical dependency popup):
File:
```
/home/gpc-610/builds/o3de/Code/Framework/AzCore/./AzCore/EBus/Internal/CallstackEntry.h
```
Condition:
```
(140659235214912) 'AZ::Internal::CallstackEntry<AzToolsFramework::ToolsApplicationRequests, AzToolsFramework::ToolsApplicationRequests>::CallstackEntry(AZ::Internal::CallstackEntry::BusContextPtr, const typename Traits::BusIdType *) [Interface = AzToolsFramework::ToolsApplicationRequests, Traits = AzToolsFramework::ToolsApplicationRequests]'
Bus static const char *AZ::EBus<AzToolsFramework::ToolsApplicationRequests>::GetName() [Interface = AzToolsFramework::ToolsApplicationRequests, BusTraits = AzToolsFramework::ToolsApplicationRequests] has multiple threads in its callstack records. Configure MutexType on the bus, or don't send to it from multiple threads
```
Line: 86

It should be noted that this issue also occurs on the Development (73e54c6) build.

**Steps to reproduce**
1. Create two entities and create two separate prefabs from them.
2. Create another two entities.
3. Duplicate one of the prefabs created in step 1.
4. Select an entity and parent it to a prefab using the *Transform* > *Parent entity* > *Pick an object in the viewport* button.
5. Repeat step 4 but parent duplicated prefab to it's original.
6. Repeat step 4 but parent a prefab to another prefab.

**Expected behavior**
Parenting to prefabs is not possible (due to https://github.com/o3de/o3de/pull/14411).

**Actual behavior**
Parenting to prefabs using the *Transform* component is possible and doing so produces multiple warnings, errors and potentially an assert.

**Video and Editor.log**

https://user-images.githubusercontent.com/86952082/226882430-a044acf9-2d57-4134-bd85-08a8621216ca.mp4

[Editor.log](https://github.com/o3de/o3de/files/11039201/Editor.log)

**Found in Branch**
Stabilization/2305 (169ab99)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
169ab99d9cad2899f99db48a12fd9bf2450a637f

**Desktop**
 - Device: PC
 - OS: Linux
 - Version: Ubuntu 22.04.1 LTS
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by raktalen** [comment]
I tried every variation of this that I could think of with latest. It seems like the editor , specifically the outliner has changed significantly since this bug was filed. At this time, if you try these operations in the editor then you get this error message. I've tried making multiple prefabs, trying to open one before clicking the other one, actually clicking in the viewport, all seem to be fine

![image](https://user-images.githubusercontent.com/118857486/228618228-73b291a0-9190-4c0b-a302-18565b3dab12.png)

**Comment by LB-KacperSzczepanski** [comment]
Reopening the issue as it still occurs on Development (0c6724e) and 22.10.0 Release installer, unless the Prefab Instance is closed. However trying to add a new entity to closed entity and then deleting, one of entities from the prefab causes an editor crash. Besides that, when a new entity is created in the level the Prefab instance automatically opens. Should we report those as separate issues?

https://user-images.githubusercontent.com/86953282/228849356-07ea53cd-64ce-4e0a-a1be-ca45d7908090.mp4

[crash_log.log](https://github.com/o3de/o3de/files/11111985/crash_log.log)
[error_dmp.zip](https://github.com/o3de/o3de/files/11112001/error_dmp.zip)

**Comment by raktalen** [comment]
gonna check if the latest editor is significantly ahead of the version you mentioned

**Comment by LB-ArturZieba** [comment]
This issue still occurs on Development (515f0cc) on both Linux and Windows:

https://user-images.githubusercontent.com/86952082/236225700-8e83d939-d71a-4c05-930c-3d7b8442e994.mp4


Should this issue be reopened or should I open a new ticket for the Development branch?

**Comment by forkercat** [comment]
Yeah, I reopened the issue.

**Comment by forkercat** [comment]
@LB-ArturZieba Hey Artur, it should be fixed now. Let me know if it isn't.

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (4e1116d) build. Attempting to parent entities/prefabs to prefabs using Transform component now shows an "Invalid Assignment" popup to the user.
